src/relay/transforms/simplify_expr.cc

```diff
@@ -708,6 +708,50 @@ class SimplifyRSqrt : public DFPatternRewrite {
   DFPattern numerator_;
 };
 
+class SimplifyDQArgFunc : public DFPatternRewrite {
+  /*! Base class for simplifying dequantize followed by arg ops */
+ public:
+  explicit SimplifyDQArgFunc(std::string op) : op_(op) {
+    x_ = IsWildcard();
+    dq_ = IsOp("qnn.dequantize")({x_, IsWildcard(), IsWildcard()});
+    pattern_ = IsOp(op_)({dq_});
+  }
+
+  Expr Callback(const Expr& pre, const Expr& post,
+                const Map<DFPattern, Array<Expr>>& node_map) const override {
+    const CallNode* call = pre.as<CallNode>();
+    ICHECK(call);
+    auto x = node_map[x_][0];
+    return Call(Op::Get(op_), {x}, call->attrs);
+  }
+
+ protected:
+  /*! \brief Pattern input */
+  DFPattern x_;
+  /*! \brief dequantize op */
+  DFPattern dq_;
+  /*! \brief Name of op to simplify */
+  String op_;
+};
+
+class SimplifyDQArgMax : public SimplifyDQArgFunc {
+  /*! Simplify dequantize follwed by argmax */
```

Comment:
Probably want to follow format used in rest of file:

```

/*! \brief Make two consecutive add able to be constant_folded.
 * This pattern matching supports commutative property for addition.
 */
class SimplifyConsecutiveAdd : public DFPatternRewrite {
```

Reply:
doh!